qiskit/primitives/backend_estimator_v2.py

```diff
@@ -178,30 +225,27 @@ def _run_pub(self, pub: EstimatorPub) -> PubResult:
         data_bin = data_bin_cls(evs=evs, stds=stds)
         return PubResult(data_bin, metadata={"target_precision": pub.precision})
 
-    def _calc_expval_paulis(
+    def _generate_circuits(
         self,
         circuit: QuantumCircuit,
         parameter_values: BindingsArray,
         param_obs_map: dict[tuple[int, ...], set[str]],
-        shots: int,
-    ) -> dict[tuple[tuple[int, ...], str], tuple[float, float]]:
-        # generate circuits
+    ) -> list[QuantumCircuit]:
```

Comment:
Suggesting a docstring to help future readers:

```suggestion
    ) -> list[QuantumCircuit]:
        """Bind the given circuit against each parameter value set, and add necessary measurements to each.
        
        Args:
            circuit: The (possibly parametric) circuit of interest.
            parameter_values: An array of parameter value sets that can be applied to the circuit.
            param_obs_map: A mapping from locations in ``parameter_values`` to a sets of 
                Pauli terms whose expectation values are required in those locations.
                
        Returns:
            A flat list of circuits sufficient to measure all Pauli terms in the ``param_obs_map`` values at the
            corresponding ``parameter_values`` location, where requisite book-keeping is stored as 
            circuit metadata.
        """
```